rasa/shared/core/training_data/story_reader/story_step_builder.py

```diff
@@ -143,9 +159,7 @@ def _next_story_steps(self) -> List[StoryStep]:
         ]
         return current_turns
 
-    def _generate_checkpoint_name_for_or_statement(
-        self, messages: List[UserUttered]
-    ) -> str:
+    def _generate_checkpoint_name_for_or_statement(self, messages: List[Event]) -> str:
```

Comment:
This method refers to `messages_texts_or_intents` - does this work for all events?

Reply:
The `__str__` method is definied for all the events so it does because it requires only that. However, the names should be changed.